mogc_partnerships/views.py

```diff
@@ -192,6 +192,77 @@ def get_serializer_context(self):
         return {"cohort": cohort}
 
 
+class CohortMembershipUpdateView(generics.UpdateAPIView):
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [IsAuthenticated]
+    serializer_class = serializers.CohortMembershipSerializer
+
+    def get_queryset(self):
+        return CohortMembership.objects.filter(
+            pk=self.kwargs.get("pk"), cohort__uuid=self.kwargs.get("cohort_uuid")
+        )
+
+    def get_object(self):
+        queryset = self.get_queryset()
+        return get_object_or_404(queryset)
+
+    def unenroll(self, cohort_member):
```

Comment:
This function feels clunky / makes me think I'm missing Djangoish things that could be used here - thoughts?